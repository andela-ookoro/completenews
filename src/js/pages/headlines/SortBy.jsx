import React from 'react';
import PropTypes from 'prop-types';

// class to display sortBy
const SortBy = ({ data, onClick }) => (
  <button value={data} onClick={onClick}> {data} </button>
);
SortBy.propTypes = {
  data: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
export default SortBy;
