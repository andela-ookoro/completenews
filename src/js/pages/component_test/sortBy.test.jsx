import React from 'react';
import { shallow, mount } from 'enzyme'; 
import Renderer from 'react-test-renderer';
import SortBy from '../headlines/SortBy';

test('Component render a button', () => {
  const data = 'test';
  const funtest = () => {
    return 1;
  };
  const component = Renderer.create(
    <SortBy data={data} source={data} onClick={funtest} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Value of render button should be same as props', () => {
  const data = "Top";
  const funtest = (e) => {
    return e.target.value;
  };
  const wrapper = mount(
    <SortBy data={data} source={data} onClick={funtest} />,
  );
  const button = wrapper.find('button');
  //console.log(button);
  expect(button.text()).toEqual(data);
});

test('Button onclick should execute the function in the prop', () => {
  const data = 'Top';
  const onClick = jest.fn();
  const wrapper = mount(
    <SortBy data={data} source={data} onClick={onClick} />,
  );
  const button = wrapper.find('button');
  button.simulate('click').tobe;
  expect(onClick).toBeCalledWith(data);
})
