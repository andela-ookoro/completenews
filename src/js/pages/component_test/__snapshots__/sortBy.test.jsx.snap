// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render a button 1`] = `
<button
  onClick={[Function]}
  value="test"
>
   
  test
   
</button>
`;

exports[`Link changes the class when hovered 1`] = `
<button
  onClick={[Function]}
  value="test"
>
   
  test
   
</button>
`;
